src/figuresClasses.ts

```diff
@@ -2,10 +1,85 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('all sides of triangle must be greater than 0');
+    }
+
+    const biggestSide = Math.max(a, b, c);
+    const sumOfSmaller = a + b + c - biggestSide;
+
+    if (biggestSide >= sumOfSmaller) {
+      throw new Error(`sides ${a}, ${b} and ${c} can not form a triangle`);
+    }
+  }
+
+  getArea(): number {
```

Comment:
The `getArea` function should not accept any parameters as it can access `a`, `b`, and `c` directly from the object's fields.